### PR TITLE
Release/15.4.0

### PR DESCRIPTION
**Pull request by duncan-lau**

### CODE REVIEW

File: PointSDK.iOS/ApiDefinition.cs [hunk 1]
```diff
@@ -13,13 +13,15 @@ namespace PointSDK.iOS
     [BaseType(typeof(NSObject))]
     interface IBDPTempoTrackingDelegate
     {
-        // @required -(void)didStartTracking;
-        [Abstract]
+        // @optional -(void)tempoTrackingDidExpire;
+        [Export("tempoTrackingDidExpire")]
+        void TempoTrackingDidExpire();
+
+        // @optional -(void)didStartTracking __attribute__((deprecated("First deprecated in 15.4.0 - This is now deprecated in favor of completion callback. Please refer to `BDLocationManager.-startTempoTrackingWithDestinationId:completion:`")));
```
Comment (poborin) [inline PointSDK.iOS/ApiDefinition.cs:20]: we need to add deprecation attributes to old API methods such as
```csharp
[Obsolete("Method1 is deprecated, please use Method2 instead.")]
```

File: PointSDK.iOS/ApiDefinition.cs [hunk 2]
```diff
@@ -34,93 +36,133 @@ interface IBDPTempoTrackingDelegate
...
     }
 
+    // @protocol BDPBluedotServiceDelegate <NSObject>
+    [Protocol, Model]
+    [BaseType(typeof(NSObject))]
+    interface IBDPBluedotServiceDelegate
+    {
+        // @optional -(void)locationAuthorizationDidChangeFromPreviousStatus:(CLAuthorizationStatus)previousAuthorizationStatus toNewStatus:(CLAuthorizationStatus)newAuthorizationStatus;
+        [Export("locationAuthorizationDidChangeFromPreviousStatus:toNewStatus:")]
+        void LocationAuthorizationDidChangeFromPreviousStatus(CLAuthorizationStatus previousAuthorizationStatus, CLAuthorizationStatus newAuthorizationStatus);
```
Comment (poborin) [inline PointSDK.iOS/ApiDefinition.cs:115]: I think we need to add a ticket to our backlog to copy API documentation from the iOS SDK project. At the moment it's quite disjointed and a developer needs to refer to the our current API documentation to find out more about methods. We should have something like this:
```csharp
        /// <summary>
        /// This method can be implemented to keep track of changes in Location Manager’s CLAuthorizationStatus.
        /// This method is called whenever location authorization status changes.
        /// You may wish to communicate to your user when location authorization status changes. For example, you may wish to ask for “Always” or “When in Use” authorization from within your app.
        /// </summary>
        /// <param name="previousAuthorizationStatus">Previous authorization status</param>
        /// <param name="newAuthorizationStatus">Current authorization status</param>
        [Export("locationAuthorizationDidChangeFromPreviousStatus:toNewStatus:")]
        void LocationAuthorizationDidChangeFromPreviousStatus(CLAuthorizationStatus previousAuthorizationStatus, CLAuthorizationStatus newAuthorizationStatus);
```

**Review by poborin — COMMENTED** [review]
I think we just need to add `[Obsolete("description")]` annotations to deprecated methods and it's good to go!

File: PointSDK.iOS/ApiDefinition.cs (hunk 1 above)
Comment (duncan-lau) [reply PointSDK.iOS/ApiDefinition.cs:20]: Added Obsolete attribute

File: PointSDK.iOS/ApiDefinition.cs (hunk 2 above)
Comment (duncan-lau) [reply PointSDK.iOS/ApiDefinition.cs:115]: Ticket created in backlog

**Review by poborin — APPROVED** [review]

**Pull request merged**